File: python/src/commands/rpush.py
from .base import BaseCommand
from src.resp.serializer import RespSerializer

class RPush(BaseCommand):
  @classmethod
  def command(cls) -> bytes:
    return b'rpush'
  
  def run(self) -> bytes:
    if len(self.args) < 2:
      return b'ERR wrong number of arguments for command'
    
    key: bytes = self.args[0]
    values: list[bytes] = self.args[1:]
    store_list: list[bytes] = self.store.get(key, [])
    if not isinstance(store_list, list):
      return b'-ERR Invalid datatype'

    for value in values:
      store_list.append(value)
    
    self.store[key] = store_list
    return RespSerializer.serialize(data=len(self.store[key]))

File: ruby/lib/redis/commands/rpush.rb
require_relative './base'

module Redis
  module Commands
    class RPush < Base
      def self.command
        'rpush'
      end

      def run
        key = args[0]
        values = args[1..]
        if key.empty? || values.empty?
          return '-ERR wrong number of arguments for command'
        end

        @store[key] ||= []
        @mutex.synchronize do
          values.each do |value|
            @store[key] = @store[key] << value
          end
        end

        Redis::Resp::Serializer.serialize(@store[key].size)
      end
    end
  end
end